src/lib/components/header/header-dashboard/HeaderDashboardDropdown.svelte

```diff
@@ -112,6 +117,50 @@
     openCreateWorkspaceModal = !openCreateWorkspaceModal;
   };
 
+  let sharedData = [];
+  function createSetFromArray(arr, key) {
+    const seen = new Set();
+    return arr.filter((obj) => {
+      if (!obj.hasOwnProperty(key)) {
+        throw new Error(`Object does not have key "${key}"`);
+      }
+      const keyValue = obj[key];
+      return !seen.has(keyValue) && seen.add(keyValue);
+    });
+  }
+
+  const calculateLimitedWorkspace = () => {
+    let workspaces = allworkspaces
+      .filter((elem) => {
+        if (currentTeam.id === elem.team.teamId) return true;
+        return false;
+      })
+      .reverse()
+      .slice(0, workspaceLimit)
+      .map((workspace) => {
+        const workspaceObj = {
+          id: workspace.name,
```

Comment:
dont we have "id" key of workspace? coz i can see we are assigning id as the name key